mongo/errors.go

```diff
@@ -27,6 +27,9 @@ var ErrClientDisconnected = errors.New("client is disconnected")
 // ErrNilDocument is returned when a nil document is passed to a CRUD method.
 var ErrNilDocument = errors.New("document is nil")
 
+// ErrMapForOrderedArgument is returned when a map with multiple keys is passed to a CRUD method for an ordered parameter
+var ErrMapForOrderedArgument = errors.New("multi-key map passed in for ordered parameter")
```

Comment:
see other comment

Reply:
I understand the existing context for `ErrNilDocument`, but I'm hesitant to add another such sentinel error just because `ErrNilDocument` already exists. It would involve a lot of code change, but would this be a viable solution that lets us insert parameter names into the error:
```
type ErrMapForOrderedArgument struct {
    paramName string
}
func (e ErrMapForOrderedArgument) Error() string {
    // same string that we have now in the sentinel value but with paramName added to it
}

func transformBsoncoreDocument(registry, val, paramName, mapsAllowed) {
    // assume buffer has bytes for transformed value
    
    if !mapsAllowed and valIsMapWithMultipleKeys() {
        return nil, ErrMapForOrderedArgument{paramName}
    }
}
```

Also, we should file a ticket with fixVersion 2.0 to do something similar for `ErrNilDocument`.

Reply:
done. created GODRIVER-1854

Also added another transformBsoncoreDocument function for MapAllowed values to hopefully make code cleaner, I can change it back if people don't think its helpful